File: it.java
import java.io.*;
import java.net.*;

public class SimpleProxyServer {
    public static void main(String[] args) throws IOException {
        int port = 8888; // Port to listen on
        ServerSocket listener = new ServerSocket(port);
        System.out.println("Proxy server started on port " + port);

        while (true) {
            Socket client = listener.accept();
            new Thread(() -> handleClient(client)).start(); // Handle client request in a separate thread
        }
    }

    private static void handleClient(Socket client) {
        try (
            InputStream clientIn = client.getInputStream();
            OutputStream clientOut = client.getOutputStream();
        ) {
            Socket server = new Socket("example.com", 80); // Connect to destination server (example.com:80)
            InputStream serverIn = server.getInputStream();
            OutputStream serverOut = server.getOutputStream();

            new Thread(() -> {
                try {
                    byte[] buffer = new byte[4096];
                    int bytesRead;
                    while ((bytesRead = clientIn.read(buffer)) != -1) {
                        serverOut.write(buffer, 0, bytesRead);
                        serverOut.flush();
                    }
                } catch (IOException e) {
                    e.printStackTrace();
                }
            }).start();

            byte[] buffer = new byte[4096];
            int bytesRead;
            while ((bytesRead = serverIn.read(buffer)) != -1) {
                clientOut.write(buffer, 0, bytesRead);
                clientOut.flush();
            }

        } catch (IOException e) {
            e.printStackTrace();
        } finally {
            try {
                client.close();
            } catch (IOException e) {
                e.printStackTrace();
            }
        }
    }
}

File: it.go
package main

import (
	"fmt"
	"io"
	"net"
	"net/http"
)

func main() {
	port := ":8888" // Port to listen on
	fmt.Printf("Proxy server started on port %s\n", port)

	// Start proxy server
	listener, err := net.Listen("tcp", port)
	if err != nil {
		fmt.Printf("Error starting server: %v\n", err)
		return
	}

	for {
		client, err := listener.Accept()
		if err != nil {
			fmt.Printf("Error accepting connection: %v\n", err)
			continue
		}
		go handleClient(client)
	}
}

func handleClient(client net.Conn) {
	defer client.Close()

	// Read client's request
	request, err := http.ReadRequest(bufio.NewReader(client))
	if err != nil {
		fmt.Printf("Error reading request: %v\n", err)
		return
	}

	// Print request details (optional)
	fmt.Printf("Request received: %s %s\n", request.Method, request.URL)

	// Connect to destination server
	server, err := net.Dial("tcp", request.URL.Host)
	if err != nil {
		fmt.Printf("Error connecting to server: %v\n", err)
		return
	}
	defer server.Close()

	// Forward client's request to server
	err = request.Write(server)
	if err != nil {
		fmt.Printf("Error forwarding request: %v\n", err)
		return
	}

	// Copy server's response back to client
	_, err = io.Copy(client, server)
	if err != nil {
		fmt.Printf("Error copying response: %v\n", err)
		return
	}

	fmt.Println("Request processed successfully")
}
